src/background/openURLs.js

```diff
@@ -53,12 +52,23 @@ openURLs.getFilter = identifiers => {
 };
 
 openURLs.handleRequest = details => {
-    const refresh = details.url.indexOf(openURLs.identifiers.refreshTab) >= 0;
-    const url = details.url.replace(`${openURLs.identifiers[refresh ? "refreshTab" : "openTab"]}/`, "");
+    const url = new URL(details.url);
+    const refresh = url.hostname.indexOf(openURLs.identifiers.refreshTab) >= 0;
+    // URL will look like: "http://opensametab.morphic.org/redirect/https%3A%2F%2Fexample.com%2F"
+    const destination = url.pathname.replace(/^\/(redirect\/)?(.*)/, (match, c1, c2) => decodeURIComponent(c2));
 
-    openURLs.openTab(url, refresh);
+    let destinationUrl;
+    try {
+        destinationUrl = new URL(destination);
+    } catch (e) {
+        // ignored
```

Comment:
I wonder if it might be helpful to log the error in a meaningful way. It likely won't make a difference for most end users, but for someone who is developing and runs into this case, it will hopefully help track things down quicker.